contrib/icons/meson.build

```diff
@@ -22,6 +22,21 @@ install_data(
     rename: 'dosbox-staging.png',
     install_dir: icons_hicolor / '32x32' / 'apps',
 )
+install_data(
+    'png' / 'icon_48.png',
+    rename: 'dosbox-staging.png',
+    install_dir: icons_hicolor / '48x48' / 'apps',
+)
+install_data(
+    'png' / 'icon_96.png',
+    rename: 'dosbox-staging.png',
+    install_dir: icons_hicolor / '96x96' / 'apps',
+)
+install_data(
+    'png' / 'icon_256.png',
+    rename: 'dosbox-staging.png',
+    install_dir: icons_hicolor / '256x256' / 'apps',
+)
```

Comment:
Looking at the meson documentation, you should be able to do something like
```meson
install_data(
    ['png/icon_32.png', 'png/icon_48.png', 'png/icon_96.png', 'png/icon_256.png'],
    rename: ['32x32/apps/dosbox-staging.png', '48x48/apps/dosbox-staging.png', '96x96/apps/dosbox-staging.png', '256x256/apps/dosbox-staging.png'],
    install_dir: icons_hicolor
)
```
Or perhaps a loop could be used?

Reply:
There was some looping previously but I removed it. My bad 😄 

Any reason not to add 512 and 1024? That would add just a few Ks. Or are sizes about 256 not supported on Linux?

Reply:
There are no 512 and 1024 sized pngs checked into the repo.  I suppose I could generate them and check them in with this PR.  I don't think the icon Makefile ever gets run except when we manually run it.  This meson.build just copies the images.

1024X1024 didn't work on my system but that's going to depend on desktop environment and possibly display settings as well.  As I'm on a 1080p monitor, it's not surprising that it didn't try to load such a large icon.

Reply:
>  Yes, and that's explained in the README in contrib/icon. I also provided you with steps a few days ago on how to add the 512 & 1024 icons to the repo, remember? 😄

Yes, I remember and I was just looking over that README as well.  I thought someone had mentioned the icon Makefile was used somewhere in our normal build system (or in CI) but I'm not seeing where if that's the case.

Either way, I'll add the larger sizes real quick.

Reply:
512X512  and 1024X1024 icons have been generated + added to the Makefile.  Also changed the meson.build to use a loop + a dict.  Looks like a cleaner solution to me  but I had to look up syntax for Meson loops so I'm doing this on the fly :wink:

Reply:
> I thought someone had mentioned the icon Makefile was used somewhere in our normal build system (or in CI) but I'm not seeing where if that's the case.

I was that, `make install` is used for the Linux packaging stuff, and that's it.